File: js/singleton.js
var Singleton = (function() {
    var instance;

    function createInstance() {
      var object = new Object("I am the instance");
    }

    return {
      getInstance: function() {
        if(!instance) {
          instance = createInstance();
        }
        return instance;
      }
    }
})();


function run() {
  var instance1 = Singleton.getInstance();
  var instance2 = Singleton.getInstance();
  alert(instance1 === instance2);
}

File: python/Singleton.py
class Singleton:
  class __Inner:
    def __init__(self,arg):
        self.value = arg
    def __str__(self):
        return repr(self)

  instance = None
  def __init__(self , arg):
      Singleton.instance = Singleton.__Inner(arg)

  def  getInstance(self,arg):
       return Singleton.instance

x = Singleton("test")
print(x.getInstance("test"))
print(x.getInstance("test"))
print(x.getInstance("test"))

